tests/test_dump_to_ckan.py

```diff
@@ -0,0 +1,370 @@
+        assert len(requests) == 3
+        assert requests[0].url == package_create_url
+        assert requests[1].url == resource_create_url
+        assert requests[2].url == resource_create_url
+
+    @requests_mock.mock()
+    def test_dump_to_ckan_package_create_streaming_resource_fail(self,
+                                                                 mock_request):
+        '''Create package with streaming resource, which failed to create
+        resource.'''
+
+        base_url = 'https://demo.ckan.org/api/3/action/'
+        package_create_url = '{}package_create'.format(base_url)
+        resource_create_url = '{}resource_create'.format(base_url)
+
+        mock_request.post(package_create_url,
+                          json={
+                            'success': True,
+                            'result': {'id': 'ckan-package-id'}})
+        mock_request.post(resource_create_url,
+                          json={
+                            'success': False,
+                            'error': {"__type": "Validation Error",
+                                      "name": ["Some validation error."]}
+                            })
+
+        # input arguments used by our mock `ingest`
+        datapackage = {
+            'name': 'my-datapackage',
+            'project': 'my-project',
+            'resources': [{
+                "dpp:streamedFrom": "https://example.com/file.csv",
+                "dpp:streaming": True,
+                "name": "resource_streamed.csv",
+                "path": "data/file.csv",
+                'schema': {'fields': [
+                    {'name': 'first', 'type': 'string'},
+                    {'name': 'last', 'type': 'string'}
+                ]}
+            }, {
+                "dpp:streamedFrom": "https://example.com/file_02.csv",
+                "name": "resource_not_streamed.csv",
+                "path": "."
+            }]
+        }
+        params = {
+            'ckan-host': 'https://demo.ckan.org',
+            'ckan-api-key': 'my-api-key',
+            'overwrite_existing': True,
+            'force-format': True
+        }
+
+        # Path to the processor we want to test
+        processor_dir = \
+            os.path.dirname(datapackage_pipelines_ckan.processors.__file__)
+        processor_path = os.path.join(processor_dir, 'dump/to_ckan.py')
+
+        # Trigger the processor with our mock `ingest` and capture what it will
+        # returned to `spew`.
+        json_file = {'first': 'Fred', 'last': 'Smith'}
+        json_file = json.dumps(json_file)
+        spew_args, _ = mock_dump_test(
+            processor_path,
+            (params, datapackage,
+             iter([ResourceIterator(io.StringIO(json_file),
+                                    datapackage['resources'][0],
+                                    {'schema': {'fields': []}})
+                   ])))
+
+        spew_res_iter = spew_args[1]
+        with self.assertRaises(Exception):
```

Comment:
Could we use a more specific exception here?

Reply:
I would be good if datapackage-pipelines had an exception class. I suppose I could use `RuntimeError`?